File: angular-frontend/src/app/common/order-item.ts
import { CartItem } from "./cart-item";

export class OrderItem {

    productId!: number;
    name!: string;
    unitPrice!: number;
    imageURL!: string;
    quantity!: number;

    constructor(cartItem: CartItem) {
        this.productId = cartItem.id;
        this.name = cartItem.name;
        this.unitPrice = cartItem.unitPrice;
        this.imageURL = cartItem.imageURL;
        this.quantity = cartItem.quantity
    }
}

File: spring-backend/src/main/java/dev/staniszak/spring/rest_jpa/interactive_webiste/entity/OrderItem.java
package dev.staniszak.spring.rest_jpa.interactive_webiste.entity;

import java.math.BigDecimal;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name="order_item")
@Getter
@Setter
public class OrderItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private long id;

    @Column(name = "name")
    private String name;

    @Column(name = "image_url")
    private String imageURL;

    @Column(name = "quantity")
    private int quantity;

    @Column(name = "unit_price")
    private BigDecimal unitPrice;

    @Column(name = "product_id")
    private long productId;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "order_id")
    private Order order;

    public OrderItem() {
    }

}
